### PR TITLE
Discord integration

### LINKED ISSUE #5251: Support Discord Intergration with AlertmanagerConfig CRD
**What is missing?**

Alertmanager version `0.25` is released with [Discord](https://discord.com/) integration. It would be nice to support that in prometheus-operator

https://github.com/prometheus/alertmanager/pull/2948

**Why do we need it?**

Feature Parity

**Environment**

* Prometheus Operator version:

    `Insert image tag or Git SHA here`
    
    

NA

**Anything else we need to know?**:

discord integration is not yet documented yet in alertmanager documentation but https://promlabs.com/blog/2022/12/23/sending-prometheus-alerts-to-discord-with-alertmanager-v0-25-0 gives overview on it

### PR DESCRIPTION
**Pull request by Azanul**
## Description
Prometheus Alertmanager brings the functionality of pushing notifications to Discord using webhooks.
Fixes #5251 

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [X] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
CRD extension for notifications on Discord
```

### CODE REVIEW

File: pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go [hunk 1]
```diff
@@ -158,6 +158,8 @@ type Receiver struct {
 	OpsGenieConfigs []OpsGenieConfig `json:"opsgenieConfigs,omitempty"`
 	// List of PagerDuty configurations.
 	PagerDutyConfigs []PagerDutyConfig `json:"pagerdutyConfigs,omitempty"`
+	// List of Discord configurations.
```
Comment (simonpasquier) [inline pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go:161]: (nit)
```suggestion
	// List of Discord configurations.
	// +optional
```

File: pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go [hunk 2]
```diff
@@ -256,6 +258,27 @@ type PagerDutyLinkConfig struct {
...
+// See https://prometheus.io/docs/alerting/latest/configuration/#discord_config
+type DiscordConfig struct {
+	// Whether or not to notify about resolved alerts.
+	// +optional
+	SendResolved *bool `json:"sendResolved,omitempty"`
+	// The secret's key that contains the Discord webhook URL.
+	// The secret needs to be in the same namespace as the AlertmanagerConfig
+	// object and accessible by the Prometheus Operator.
+	// +optional
+	APIURL *v1.SecretKeySelector `json:"apiURL,omitempty"`
```
Comment (simonpasquier) [inline pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go:271]: this should be a required field? IIUC there's no global Discord URL parameter and Alertmanager will fail to load a discord config with an empty URL.

**Review by simonpasquier — COMMENTED** [review]
Thanks! I've only looked at the API additions for now.

File: pkg/alertmanager/amcfg.go [hunk 3]
```diff
@@ -513,6 +512,18 @@ func (cb *configBuilder) convertReceiver(ctx context.Context, in *monitoringv1al
 		}
 	}
 
+	var discordConfigs []*discordConfig
+	if l := len(in.DiscordConfigs); l > 0 {
+		discordConfigs = make([]*discordConfig, l)
+		for i := range in.SlackConfigs {
+			receiver, err := cb.convertDiscordConfig(ctx, in.DiscordConfigs[i], crKey)
+			if err != nil {
+				return nil, errors.Wrapf(err, "SlackConfig[%d]", i)
```
Comment (JoaoBraveCoding) [inline pkg/alertmanager/amcfg.go:521]:
```suggestion
				return nil, errors.Wrapf(err, "DiscordConfig[%d]", i)
```

File: pkg/alertmanager/amcfg.go [hunk 4]
```diff
@@ -513,6 +512,18 @@ func (cb *configBuilder) convertReceiver(ctx context.Context, in *monitoringv1al
 		}
 	}
 
+	var discordConfigs []*discordConfig
+	if l := len(in.DiscordConfigs); l > 0 {
+		discordConfigs = make([]*discordConfig, l)
+		for i := range in.SlackConfigs {
```
Comment (JoaoBraveCoding) [inline pkg/alertmanager/amcfg.go:518]:
```suggestion
		for i := range in.DiscordConfigs {
```

**Review by JoaoBraveCoding — COMMENTED** [review]
Just two comments, but overall looks good! Would you mind adding some unit tests to `amcfg_test.go`

**Review by ArthurSens — COMMENTED** [review]
Adding a test and addressing Simon's comment and it should be good!

Thanks for the contribution :)

File: pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go (hunk 2 above)
Comment (Azanul) [reply pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go:272]: Ref: https://prometheus.io/docs/alerting/latest/configuration/#configuration-file

Comment (ArthurSens) [reply pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go]: Strange, I can't seem to find discord configs in the docs, although I can see the support by reading Alertmanager's codebase.

Comment (ArthurSens) [reply pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go]: I guess the documentation on the website only shows the global config, and indeed Discord is not listed there:

https://github.com/prometheus/alertmanager/blob/a85979e19d24490322d5ce342301d17b0f13dcc5/config/config.go#L737-L767

https://github.com/prometheus/alertmanager/blob/a85979e19d24490322d5ce342301d17b0f13dcc5/config/config.go#L628-L644

File: pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go [hunk 5]
```diff
@@ -256,6 +259,26 @@ type PagerDutyLinkConfig struct {
...
+// DiscordConfig configures notifications via Discord.
+// See https://prometheus.io/docs/alerting/latest/configuration/#discord_config
+type DiscordConfig struct {
+	// Whether or not to notify about resolved alerts.
+	// +optional
+	SendResolved *bool `json:"sendResolved,omitempty"`
+	// The secret's key that contains the Discord webhook URL.
+	// The secret needs to be in the same namespace as the AlertmanagerConfig
+	// object and accessible by the Prometheus Operator.
+	APIURL v1.SecretKeySelector `json:"apiURL,omitempty"`
```
Comment (ArthurSens) [inline pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go:271]:
```suggestion
	APIURL v1.SecretKeySelector `json:"apiURL"`
```

If you keep `omitempty`, it is still optional. I agree that it is quite confusing 😅

Comment (Azanul) [reply pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go:271]: I won't say confusing but it's like we have a checklist 📝 for converting optional fields to mandatory😅

File: pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go (hunk 2 above)
Comment (Azanul) [reply pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go]: https://github.com/prometheus/alertmanager/issues/3360

It seems like it was supposed to be released but hasn't been yet.

File: pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go [hunk 6]
```diff
@@ -256,6 +259,26 @@ type PagerDutyLinkConfig struct {
...
+// See https://prometheus.io/docs/alerting/latest/configuration/#discord_config
+type DiscordConfig struct {
+	// Whether or not to notify about resolved alerts.
+	// +optional
+	SendResolved *bool `json:"sendResolved,omitempty"`
+	// The secret's key that contains the Discord webhook URL.
+	// The secret needs to be in the same namespace as the AlertmanagerConfig
+	// object and accessible by the Prometheus Operator.
+	APIURL v1.SecretKeySelector `json:"apiURL"`
+	// +optional
```
Comment (simonpasquier) [inline pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go:272]:
```suggestion
	// The template of the message's title.
	// +optional
```

File: pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go [hunk 7]
```diff
@@ -256,6 +259,26 @@ type PagerDutyLinkConfig struct {
...
+	Title *string `json:"title,omitempty"`
+	// +optional
```
Comment (simonpasquier) [inline pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go:274]:
```suggestion
	// The template of the message's body.
	// +optional
```

File: pkg/alertmanager/amcfg.go [hunk 8]
```diff
@@ -670,6 +682,30 @@ func (cb *configBuilder) convertWebhookConfig(ctx context.Context, in monitoring
 	return out, nil
 }
 
+func (cb *configBuilder) convertDiscordConfig(ctx context.Context, in monitoringv1alpha1.DiscordConfig, crKey types.NamespacedName) (*discordConfig, error) {
+	out := &discordConfig{
+		VSendResolved: in.SendResolved,
+		Title:         *in.Title,
+		Message:       *in.Message,
+	}
```
Comment (simonpasquier) [inline pkg/alertmanager/amcfg.go:690]: you need to check for nil

```suggestion
	out := &discordConfig{
		VSendResolved: in.SendResolved,
	}
	
	if in.Title != nil && *in.Title != "" {
		out.Title = *in.Title
	}
	
	if in.Message != nil && *in.Message != "" {
		out.Message = *in.Message
	}
```

Comment (Azanul) [reply pkg/alertmanager/amcfg.go:690]: @simonpasquier If I keep it `string` instead of `*string` then also it'll remain to be optional, right?

Comment (ArthurSens) [reply pkg/alertmanager/amcfg.go:690]: Yes, but we want to keep them as pointers to be compliant with [kubernetes conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required).

With them as pointers, we need to check for nils 😬

Comment (Azanul) [reply pkg/alertmanager/amcfg.go:690]: Thanks alot for the reference @ArthurSens

**Review by JoaoBraveCoding — COMMENTED** [review]
lgtm, just have to fix the ci, part of it is just running `make format`

**Comment by Azanul** [comment]
> lgtm, just have to fix the ci, part of it is just running `make format`

`make format` isn't changing anything.
Screenshot 2023-07-01 at 10 50 16 AM

**Comment by JoaoBraveCoding** [comment]
Apologies @Azanul maybe `make generate format`

**Comment by Azanul** [comment]
> Apologies @Azanul maybe `make generate format`

No file change, here's the log

```
$ make generate format
Installing tools from scripts/tools.go
go install '-mod=readonly' '-modfile=scripts/go.mod' github.com/ahmetb/gen-crd-api-reference-docs
go install '-mod=readonly' '-modfile=scripts/go.mod' github.com/brancz/gojsontoyaml
go install '-mod=readonly' '-modfile=scripts/go.mod' github.com/bwplotka/mdox
go install '-mod=readonly' '-modfile=scripts/go.mod' github.com/golangci/golangci-lint/cmd/golangci-lint
go install '-mod=readonly' '-modfile=scripts/go.mod' github.com/google/go-jsonnet/cmd/jsonnet
go install '-mod=readonly' '-modfile=scripts/go.mod' github.com/google/go-jsonnet/cmd/jsonnetfmt
go install '-mod=readonly' '-modfile=scripts/go.mod' github.com/jsonnet-bundler/jsonnet-bundler/cmd/jb
go install '-mod=readonly' '-modfile=scripts/go.mod' github.com/yeya24/promlinter/cmd/promlinter
go install '-mod=readonly' '-modfile=scripts/go.mod' k8s.io/code-generator/cmd/applyconfiguration-gen
go install '-mod=readonly' '-modfile=scripts/go.mod' k8s.io/code-generator/cmd/client-gen
go install '-mod=readonly' '-modfile=scripts/go.mod' k8s.io/code-generator/cmd/informer-gen
go install '-mod=readonly' '-modfile=scripts/go.mod' k8s.io/code-generator/cmd/lister-gen
go install '-mod=readonly' '-modfile=scripts/go.mod' sigs.k8s.io/controller-tools/cmd/controller-gen
Downloading shellcheck
shellcheck-stable/shellcheck
cd ./pkg/apis/monitoring/v1 && /workspaces/prometheus-operator/tmp/bin/controller-gen object:headerFile=/workspaces/prometheus-operator/.header \
        paths=.
cd ./pkg/apis/monitoring/v1alpha1 && /workspaces/prometheus-operator/tmp/bin/controller-gen object:headerFile=/workspaces/prometheus-operator/.header \
        paths=.
cd ./pkg/apis/monitoring/v1beta1 && /workspaces/prometheus-operator/tmp/bin/controller-gen object:headerFile=/workspaces/prometheus-operator/.header \
        paths=.
rm -rf pkg/client/{versioned,informers,listers,applyconfiguration}
>> generating pkg/client/applyconfiguration...
/workspaces/prometheus-operator/tmp/bin/applyconfiguration-gen \
        --go-header-file /workspaces/prometheus-operator/.header --v=1 --logtostderr \
        --input-dirs      "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1,github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1alpha1,github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1beta1" \
        --output-package  "github.com/prometheus-operator/prometheus-operator/pkg/client/applyconfiguration" \
        --output-base    "."
mv github.com/prometheus-operator/prometheus-operator/pkg/client/applyconfiguration pkg/client
>> generating pkg/client/versioned...
/workspaces/prometheus-operator/tmp/bin/client-gen \
        --go-header-file /workspaces/prometheus-operator/.header --v=1 --logtostderr \
        --input-base                  "" \
        --apply-configuration-package "github.com/prometheus-operator/prometheus-operator/pkg/client/applyconfiguration" \
        --clientset-name              "versioned" \
        --input                       "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1,github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1alpha1,github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1beta1" \
        --output-package              "github.com/prometheus-operator/prometheus-operator/pkg/client" \
        --output-base                 "."
>> generating pkg/client/listers...
/workspaces/prometheus-operator/tmp/bin/lister-gen \
        --go-header-file /workspaces/prometheus-operator/.header --v=1 --logtostderr \
        --input-dirs     "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1,github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1alpha1,github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1beta1" \
        --output-package "github.com/prometheus-operator/prometheus-operator/pkg/client/listers" \
        --output-base    "."
>> generating pkg/client/informers...
/workspaces/prometheus-operator/tmp/bin/informer-gen \
        --go-header-file /workspaces/prometheus-operator/.header --v=1 --logtostderr \
        --versioned-clientset-package "github.com/prometheus-operator/prometheus-operator/pkg/client/versioned" \
        --listers-package "github.com/prometheus-operator/prometheus-operator/pkg/client/listers" \
        --input-dirs      "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1,github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1alpha1,github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1beta1" \
        --output-package  "github.com/prometheus-operator/prometheus-operator/pkg/client/informers" \
        --output-base    "."
mv github.com/prometheus-operator/prometheus-operator/pkg/client/{versioned,informers,listers} pkg/client
rm -r github.com
cd pkg/apis/monitoring && /workspaces/prometheus-operator/tmp/bin/controller-gen crd:crdVersions=v1 paths=./v1/. paths=./v1alpha1/. output:crd:dir=/workspaces/prometheus-operator/example/prometheus-operator-crd/
find example/prometheus-operator-crd/ -name '*.yaml' -print0 | xargs -0 -I{} sh -c '/workspaces/prometheus-operator/tmp/bin/gojsontoyaml -yamltojson < "$1" | jq > "/workspaces/prometheus-operator/jsonnet/prometheus-operator/$(basename $1 | cut -d'_' -f2 | cut -d. -f1)-crd.json"' -- {}
cd pkg/apis/monitoring && /workspaces/prometheus-operator/tmp/bin/controller-gen crd:crdVersions=v1 paths=./... output:crd:dir=/workspaces/prometheus-operator/example/prometheus-operator-crd-full
echo "// Code generated using 'make generate-crds'. DO NOT EDIT." > /workspaces/prometheus-operator/jsonnet/prometheus-operator/alertmanagerconfigs-v1beta1-crd.libsonnet
echo "{spec+: {versions+: $(/workspaces/prometheus-operator/tmp/bin/gojsontoyaml -yamltojson < example/prometheus-operator-crd-full/monitoring.coreos.com_alertmanagerconfigs.yaml | jq '.spec.versions | map(select(.name == "v1beta1"))')}}" | /workspaces/prometheus-operator/tmp/bin/jsonnetfmt - >> /workspaces/prometheus-operator/jsonnet/prometheus-operator/alertmanagerconfigs-v1beta1-crd.libsonnet
cd scripts/generate; /workspaces/prometheus-operator/tmp/bin/jb install;
scripts/generate/build-rbac-prometheus-operator.sh
scripts/generate-bundle.sh
mkdir -p example/alerts
/workspaces/prometheus-operator/tmp/bin/jsonnet jsonnet/mixin/alerts.jsonnet | /workspaces/prometheus-operator/tmp/bin/gojsontoyaml > example/mixin/alerts.yaml
scripts/generate/build-thanos-example.sh
scripts/generate/build-admission-webhook-example.sh
scripts/generate/build-conversion-webhook-patch-for-alermanagerconfig-crd.sh
GOOS=linux GOARCH=arm64 CGO_ENABLED=0 go build -ldflags="-s -X github.com/prometheus/common/version.Revision=1ae80eaf0 -X github.com/prometheus/common/version.BuildUser=vscode -X github.com/prometheus/common/version.BuildDate=20230703-15:44:35 -X github.com/prometheus/common/version.Branch=discord-integration -X github.com/prometheus/common/version.Version=0.65.2" -o operator cmd/operator/main.go
/workspaces/prometheus-operator/tmp/bin/mdox fmt Documentation/operator.md
gofmt -s -w .
find . -name *.jsonnet -or -name *.libsonnet -not -path "*/vendor/*" -print0 | xargs -0 /workspaces/prometheus-operator/tmp/bin/jsonnetfmt -i
./scripts/check_license.sh
Installing tools from scripts/tools.go
go install '-mod=readonly' '-modfile=scripts/go.mod' github.com/ahmetb/gen-crd-api-reference-docs
go install '-mod=readonly' '-modfile=scripts/go.mod' github.com/brancz/gojsontoyaml
go install '-mod=readonly' '-modfile=scripts/go.mod' github.com/bwplotka/mdox
go install '-mod=readonly' '-modfile=scripts/go.mod' github.com/golangci/golangci-lint/cmd/golangci-lint
go install '-mod=readonly' '-modfile=scripts/go.mod' github.com/google/go-jsonnet/cmd/jsonnet
go install '-mod=readonly' '-modfile=scripts/go.mod' github.com/google/go-jsonnet/cmd/jsonnetfmt
go install '-mod=readonly' '-modfile=scripts/go.mod' github.com/jsonnet-bundler/jsonnet-bundler/cmd/jb
go install '-mod=readonly' '-modfile=scripts/go.mod' github.com/yeya24/promlinter/cmd/promlinter
go install '-mod=readonly' '-modfile=scripts/go.mod' k8s.io/code-generator/cmd/applyconfiguration-gen
go install '-mod=readonly' '-modfile=scripts/go.mod' k8s.io/code-generator/cmd/client-gen
go install '-mod=readonly' '-modfile=scripts/go.mod' k8s.io/code-generator/cmd/informer-gen
go install '-mod=readonly' '-modfile=scripts/go.mod' k8s.io/code-generator/cmd/lister-gen
go install '-mod=readonly' '-modfile=scripts/go.mod' sigs.k8s.io/controller-tools/cmd/controller-gen
Downloading shellcheck
shellcheck-stable/shellcheck
/workspaces/prometheus-operator/tmp/bin/shellcheck ./scripts/check_license.sh ./scripts/generate/build-admission-webhook-example.sh ./scripts/generate/build-conversion-webhook-patch-for-alermanagerconfig-crd.sh ./scripts/generate/build-non-rbac-prometheus-operator.sh ./scripts/generate/build-rbac-prometheus-operator.sh ./scripts/generate/build-thanos-example.sh ./scripts/generate-bundle.sh ./scripts/push-docker-image.sh ./scripts/run-external.sh
>> formatting and local/remote link check
/workspaces/prometheus-operator/tmp/bin/mdox fmt --soft-wraps -l --links.localize.address-regex="https://prometheus-operator.dev/.*" --links.validate.config-file=.mdox.validate.yaml Documentation/additional-scrape-config.md Documentation/custom-configuration.md Documentation/design.md Documentation/designs/prometheus-agent.md Documentation/exposing-metrics.md Documentation/high-availability.md Documentation/network-policies.md Documentation/proposals/202212-scrape-config.md Documentation/rbac-crd.md Documentation/rbac.md Documentation/thanos.md Documentation/troubleshooting.md Documentation/user-guides/alerting.md Documentation/user-guides/basic-auth.md Documentation/user-guides/exposing-prometheus-and-alertmanager.md Documentation/user-guides/getting-started.md Documentation/user-guides/linting.md Documentation/user-guides/monitoring-kubernetes-ingress.md Documentation/user-guides/prometheus-agent.md Documentation/user-guides/running-exporters.md Documentation/user-guides/shards-and-replicas.md Documentation/user-guides/storage.md Documentation/user-guides/strategic-merge-patch.md Documentation/user-guides/webhook.md CHANGELOG.md code-of-conduct.md CONTRIBUTING.md governance.md MAINTAINERS.md README.md RELEASE.md SECURITY.md
⡿ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/exposing-prometheus-and-aler
⣟ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/exposing-prometheus-and-aler
⣯ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/exposing-prometheus-and-aler
⣷ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/exposing-prometheus-and-aler
⣾ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/exposing-prometheus-and-aler
⣽ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/exposing-prometheus-and-aler
⣻ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/exposing-prometheus-and-aler
⣟ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣯ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣷ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣾ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣽ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣻ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⢿ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⡿ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣟ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣯ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣷ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣾ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣽ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣻ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⢿ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⡿ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣟ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣯ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣷ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣾ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣽ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣻ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⢿ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⡿ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣟ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣯ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣷ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣾ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣽ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣻ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⢿ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⡿ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣟ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣯ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣷ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣾ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣽ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣻ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⢿ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⡿ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣟ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣯ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣷ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣾ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣽ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣻ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⢿ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⡿ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣟ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣯ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣷ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣾ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣽ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣻ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⢿ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⡿ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣟ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣯ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣷ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣾ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣽ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣻ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⢿ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⡿ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣟ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣯ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣷ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣾ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣽ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣻ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⢿ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⡿ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣟ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣯ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣷ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣾ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣽ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣻ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⢿ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⡿ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
⣟ Formatting: /workspaces/prometheus-operator/Documentation/user-guides/monitoring-kubernetes-ingres
```

**Review by ArthurSens — APPROVED** [review]
Thanks @Azanul for the contribution and @simonpasquier for the final push :)

**Pull request merged**